### PR TITLE
VM ML bindings execution charge estimators

### PR DESCRIPTION
**Pull request by lrahmani**
Reference: 
- https://docs.google.com/document/d/1xIEYt8B-yPo0W4igpMe8Rel9hd2eU0GaCtqk6aywJBs/edit
- https://docs.google.com/document/d/16ORxnBv0BsIgdNgiGZx1UktejbENkNPg474CD-IpuJQ/edit

### CODE REVIEW

File: libs/meta/include/meta/tuple.hpp [hunk 1]
```diff
@@ -74,6 +74,21 @@ struct TupleOperations<std::tuple<Ts...>>
...
+  {
+    using type = std::tuple<Ts2..., Ts...>;
+  };
+  template <typename T>
+  struct Append;
+  template <typename... Ts2>
+  struct Append<std::tuple<Ts2...>>
+  {
+    using type = std::tuple<Ts..., Ts2...>;
+  };
```
Comment (denis-trapeznikov) [inline libs/meta/include/meta/tuple.hpp:91]: `template<class... Ts> using TupleCat = decltype(std::tuple_cat(std::declval<Ts>()...));` ?

File: libs/vm/include/vm/module/binding_interfaces.hpp [hunk 2]
```diff
@@ -37,6 +37,46 @@ namespace vm {
...
+struct PrependedTupleImpl;
+
+template <typename T, typename... Ts>
+struct PrependedTupleImpl<T, std::tuple<Ts...>>
+{
+  using type = std::tuple<T, Ts...>;
+};
+
+template <typename T, typename Tuple>
+using PrependedTuple = typename PrependedTupleImpl<T, Tuple>::type;
```
Comment (denis-trapeznikov) [inline libs/vm/include/vm/module/binding_interfaces.hpp:50]: `Cons` would be a more traditional name here. Besides, if we take `TupleCat` from my comment above, then it is trivially `template <class Car, class Cdr> using Cons = TupleCat<std::tuple<Car>, Cdr>;`.

**Review by denis-trapeznikov — CHANGES_REQUESTED** [review]
Nitpicked few places to simplify.

**Review by wkaluza — CHANGES_REQUESTED** [review]
VMTensor estimator tests were rejected in other reviews.

**Pull request closed without merging**